src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue`1.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+
+namespace Roslyn.Utilities
+{
+    /// <inheritdoc cref="AsyncBatchingWorkQueue{TItem, TResult}"/>
+    internal class AsyncBatchingWorkQueue<TItem> : AsyncBatchingWorkQueue<TItem, bool>
```

Comment:
```suggestion
    internal class AsyncBatchingWorkQueue<TItem> : AsyncBatchingWorkQueue<TItem, VoidResult>
```

Reply:
📝 If you make this change, you'll probably also want to switch from `Task<bool>` to `ValueTask<TResult>` to avoid allocating tasks when `processBatchAsync` completes synchronously.